service/orchestrator/certificates.go

```diff
@@ -0,0 +1,172 @@
+package orchestrator
+
+import (
+	"context"
+	"errors"
+
+	"clouditor.io/clouditor/api/orchestrator"
+	"clouditor.io/clouditor/internal/logging"
+	"clouditor.io/clouditor/persistence"
+	"clouditor.io/clouditor/service"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/exp/slices"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+const certificationNotFoundErrorMessage = "certificate not found"
+
+// CreateCertificate implements method for creating a new certificate
+func (svc *Service) CreateCertificate(ctx context.Context, req *orchestrator.CreateCertificateRequest) (
+	res *orchestrator.Certificate, err error) {
+
+	// Validate request
+	if err = service.ValidateRequest(req); err != nil {
+		err = status.Error(codes.InvalidArgument, err.Error())
+		return
+	}
+
+	// Check if client is allowed to access the corresponding cloud service (targeted in the certificate)
+	if !svc.authz.CheckAccess(ctx, service.AccessCreate, req) {
+		err = service.ErrPermissionDenied
+		return
+	}
+
+	// Persist the new certificate in our database
+	err = svc.storage.Create(req.Certificate)
+	if err != nil {
+		err = status.Errorf(codes.Internal, "could not add certificate to the database: %v", err)
+		return
+	}
+
+	logging.LogRequest(log, logrus.DebugLevel, logging.Create, req)
+
+	// Return certificate
+	res = req.Certificate
+	return
+}
+
+// GetCertificate implements method for getting a certificate, e.g. to show its state in the UI
+func (svc *Service) GetCertificate(ctx context.Context, req *orchestrator.GetCertificateRequest) (
+	res *orchestrator.Certificate, err error) {
+
+	// Validate request
+	if err = service.ValidateRequest(req); err != nil {
+		err = status.Error(codes.InvalidArgument, err.Error())
+		return
+	}
+
+	res = new(orchestrator.Certificate)
+	err = svc.storage.Get(res, "Id = ?", req.CertificateId)
+	if errors.Is(err, persistence.ErrRecordNotFound) {
+		return nil, status.Errorf(codes.NotFound, certificationNotFoundErrorMessage)
+	} else if err != nil {
+		return nil, status.Errorf(codes.Internal, "database error: %v", err)
+	}
+
+	// Check if client is allowed to access the corresponding cloud service (targeted in the certificate)
+	areAllAllowed, allowedCloudServices := svc.authz.AllowedCloudServices(ctx)
```

Comment:
we used the variables `all` and `allowed` in other places, would be good to be consistent.

And actually we should append the cloud service check in the query (maybe we should also have a helper function for this), instead of first querying the DB and then looking at it.

Reply:
1. Yes, i know and I don't like them :D For me they are inexpressive or even counter-intuitive: `allowed` suggests it is a boolean more than `all`. And i would prefer prefixing bools with is/has/... except they are common like `ok`
2. We could do it but then I guess we can't differentiate between NotFound and PermissionDenied anymore? Maybe there is some DB query trick I'm not aware of.

Reply:
Will be done in future PR